### DISCUSSION TITLE
Issues with credentials configuring Mend Renovate Github app with Github packages in the same organization.

### DISCUSSION THREAD

**Discussion by HalvorNygaard** [renovatebot/renovate#31307] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

I have added the Mend Renovate GitHub app to an organization, for use on .NET projects, and i have two NuGet sources. One is nuget.org, and the other one is the GitHub packages for that same organization. I have given it all the permissions it asked for when i added the application, and i have also tried adding a PAT in the host rules section in the app, and none of them seem to do the trick.

Attached the a bit of the log i thought was relevant. I see when it tries to do the package lookup it gets a 401, so I am wondering if i am missing any obvious steps in the setup of the renovate app, or other things. 

### Logs (if relevant)

<details><summary>Logs</summary>

```json
INFO: Dependency extraction complete
{
  "baseBranch": "main"
  "stats": {
    "managers": {
      "dockerfile": {
        "fileCount": 2,
        "depCount": 4
      },
      "github-actions": {
        "fileCount": 3,
        "depCount": 3
      },
      "nuget": {
        "fileCount": 2,
        "depCount": 31
      }
    },
    "total": {
      "fileCount": 7,
      "depCount": 38
    }
  }
}

DEBUG: hostRules: no authentication for api.nuget.org
DEBUG: Using queue: host=api.nuget.org, concurrency=16
DEBUG: Using queue: host=nuget.devexpress.com, concurrency=16
DEBUG: Using queue: host=nuget.pkg.github.com, concurrency=16
DEBUG: GET https://nuget.pkg.github.com/<Removed>/index.json = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=401 retryCount=0, duration=91)
DEBUG: nuget registry failure: can't get RegistrationsBaseUrl
{
  "err": {
    "name": "HTTPError",
    "code": "ERR_NON_2XX_3XX_RESPONSE",
    "timings": {
      "start": 1725961649986,
      "socket": 1725961649987,
      "lookup": 1725961650058,
      "connect": 1725961650066,
      "secureConnect": 1725961650074,
      "upload": 1725961650074,
      "response": 1725961650077,
      "end": 1725961650077,
      "phases": {
        "wait": 1,
        "dns": 71,
        "tcp": 8,
        "tls": 8,
        "request": 0,
        "firstByte": 3,
        "download": 0,
        "total": 91
      }
    },
    "message": "Response code 401 (Unauthorized)",
    "stack": "HTTPError: Response code 401 (Unauthorized)\n    at Request.<anonymous> (/usr/local/renovate/node_modules/.pnpm/got@11.8.6/node_modules/got/dist/source/as-promise/index.js:118:42)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)",
    "options": {
      "headers": {
        "user-agent": "RenovateBot/38.59.2 (https://github.com/renovatebot/renovate)",
        "accept": "application/json",
        "accept-encoding": "gzip, deflate, br"
      },
      "url": "https://nuget.pkg.github.com/<Removed>/index.json",
      "hostType": "nuget",
      "username": "",
      "password": "",
      "method": "GET",
      "http2": false
    },
    "response": {
      "statusCode": 401,
      "statusMessage": "Unauthorized",
      "body": "",
      "headers": {
        "access-control-allow-methods": "GET, OPTIONS",
        "access-control-allow-origin": "*",
        "content-security-policy": "default-src 'none';",
        "server": "GitHub Registry",
        "strict-transport-security": "max-age=31536000;",
        "www-authenticate": "Basic realm=\"GitHub Package Registry\"",
        "x-content-type-options": "nosniff",
        "x-frame-options": "DENY",
        "x-nuget-warning": "No Authorization header detected",
        "x-xss-protection": "1; mode=block",
        "date": "Tue, 10 Sep 2024 09:47:30 GMT",
        "content-length": "0",
        "x-github-request-id": "B728:3C6505:4B8622:57C6AB:66E015B2"
      },
      "httpVersion": "1.1",
      "retryCount": 0
    }
  }
  "url": "https://nuget.pkg.github.com/<Removed>/index.json"
}
```

</details>

**Comment by rarkins** [comment]
As far as I'm aware (see https://github.com/orgs/community/discussions/26920) GitHub's app tokens still do not support github packages. This means that a host rule with PAT is definitely required.

The best thing you can do is:
- Find an example of a necessary API endpoint on nuget.pkg.github.com which you need to access
- Use `curl` or similar tool with the token you have and validate that you can get a 200 OK
- If it's not obvious how to map that to a Renovate host rules, describe here how you used the token and we can advise

Normally we see either:
- Token is used as normal Bearer Authorization token, or
- Token is used as "password" and combined with github username and Basic authentication is used

**Reply by HalvorNygaard** [reply]
I used an http client to test the endpoint "https://nuget.pkg.github.com/<owner>/index.json" with the bearer token, and i got a 200 response with a response body that is a description of the API. I see that getting data for one specific package uses the URL format "https://nuget.pkg.github.com/<owner>/<package>/index.json" which also worked and gave me all the meta data from the versions of the package. 

I don't have any host rules related stuff in the renovate.config, I have it all in the GitHub App. I setup a host rule on the organization level that looks like this. (using the same token that i verified from earlier)

<img width="447" alt="image" src="https://github.com/user-attachments/assets/5d37176a-a8ae-4600-ad8a-ba37ff9c0b7b">

**Reply by rarkins** [reply, marked as answer]
hostType=github is your problem. Renovate looks for hostType=nuget in such circumstances

**Reply by HalvorNygaard** [reply]
Thanks!
